File: Inwentaryzacja/src/main/java/Zut_IPZ/Inwentaryzacja/item/Item.java
package Zut_IPZ.Inwentaryzacja.item;

import Zut_IPZ.Inwentaryzacja.tag.Tag;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.util.Date;
import java.util.HashSet;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@RequiredArgsConstructor
@Entity
@NoArgsConstructor
@Table(name="item")
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class,property="id")
public class Item {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private @NonNull String name;
    private @NonNull Boolean fillable;
    private @NonNull Date createdAt;
    private String description;

    @ManyToMany(cascade={CascadeType.PERSIST,CascadeType.MERGE})
    @JoinTable(name="tag_in_item",
            joinColumns=@JoinColumn(name="item_id"),
            inverseJoinColumns=@JoinColumn(name="tag_id"))
    private Set<Tag> tags = new HashSet<>();
}

File: testing_requirements/item.ts
interface Item {
  id: number;
  name: string;
  fillable: boolean;
  createdAt: string;
  description: string;
  tags: Tag;
}

interface Tag {
  id: number;
  attributes: Array<Attribute>;
}


interface Attribute {
  id: number;
  name: string;
  type: AttributeType;
}

interface AttributeType {
  id: number;
  name: string;
}
